### PR TITLE
Fix 'task name is not templated in retry callback' (add task_name property to TaskResult)

### LINKED ISSUE #18236: Task name is not templated in "FAILED - RETRYING: TASK:" message
##### ISSUE TYPE

- Feature Idea
##### COMPONENT NAME

[CallbackModule](https://github.com/ansible/ansible/blob/devel/lib/ansible/plugins/callback/default.py) or [StrategyBase](https://github.com/ansible/ansible/blob/devel/lib/ansible/plugins/strategy/__init__.py)

This is the method printing the message in question:
CallbackModule.v2_runner_retry
and it's called from
StrategyBase._process_pending_results
##### ANSIBLE VERSION

```
ansible 2.3.0 (devel dceb2a0393) last updated 2016/10/28 09:48:34 (GMT +200)
  lib/ansible/modules/core: (detached HEAD 124bb92416) last updated 2016/10/28 09:48:37 (GMT +200)
  lib/ansible/modules/extras: (detached HEAD 8ffe314ea5) last updated 2016/10/25 10:21:06 (GMT +200)
  config file = /etc/ansible/ansible.cfg
  configured module search path = Default w/o overrides

```
##### CONFIGURATION

retry_files_enabled = False
pipelining = True
##### OS / ENVIRONMENT

CentOS release 6.8 (Final)
##### SUMMARY

Not sure if this is a bug report or a feature request, but it would be nice to have the variables in task name templated when it's printed in "FAILED - RETRYING: TASK:" message
##### STEPS TO REPRODUCE

Put a variable in name of a task that is defined to retry until a condition is fulfilled and make it retry at least once.

```

---
- hosts: localhost
  tasks:
  - name: First run
    command: echo 1
    register: result

  - name: Previous result {{ result.stdout }}
    command: echo 2
    register: result
    retries: 1
    delay: 1
    until: False
```

##### EXPECTED RESULTS

```
PLAY [localhost] ***************************************************************

TASK [setup] *******************************************************************
ok: [localhost]

TASK [First run] ***************************************************************
changed: [localhost]

TASK [Previous result 1] *******************************************************
FAILED - RETRYING: TASK: Previous result 2 (1 retries left).
fatal: [localhost]: FAILED! => {"attempts": 1, "changed": true, "cmd": ["echo", "2"], "delta": "0:00:00.004803", "end": "2016-10-28 16:34:03.315666", "failed": true, "rc": 0, "start": "2016-10-28 16:34:03.310863", "stderr": "", "stdout": "2", "stdout_lines": ["2"], "warnings": []}

```
##### ACTUAL RESULTS

```
PLAY [localhost] ***************************************************************

TASK [setup] *******************************************************************
ok: [localhost]

TASK [First run] ***************************************************************
changed: [localhost]

TASK [Previous result 1] *******************************************************
FAILED - RETRYING: TASK: Previous result {{ result.stdout }} (1 retries left).
fatal: [localhost]: FAILED! => {"attempts": 1, "changed": true, "cmd": ["echo", "2"], "delta": "0:00:00.004803", "end": "2016-10-28 16:34:03.315666", "failed": true, "rc": 0, "start": "2016-10-28 16:34:03.310863", "stderr": "", "stdout": "2", "stdout_lines": ["2"], "warnings": []}

```

### PR DESCRIPTION
**Pull request by alikins**
Fixes #18236

##### ISSUE TYPE

 - Bugfix Pull Request

##### COMPONENT NAME

lib/ansible/executor/task_executor.py
lib/ansible/plugins/callback/default.py

##### ANSIBLE VERSION

```
ansible 2.3.0 (retry_fail_18236 b968b2964e) last updated 2017/02/09 15:46:04 (GMT -400)
  config file = /home/adrian/.ansible.cfg
  configured module search path = [u'/home/adrian/ansible/my-modules', u'/home/adrian/ansible/lib/modules/']

```

### CODE REVIEW

File: lib/ansible/executor/task_executor.py [hunk 1]
```diff
@@ -564,6 +564,9 @@ def _evaluate_failed_when_result(result):
                     if attempt < retries:
                         result['_ansible_retry'] = True
                         result['retries'] = retries
+                        # Add the original task name into the result, so TaskResult can consult it for its name
+                        result['task_name'] = wrap_var(self._task.get_name())
```
Comment (bcoca) [inline lib/ansible/executor/task_executor.py]: _ansible_task_name? to avoid collisions

Comment (bcoca) [reply lib/ansible/executor/task_executor.py]: doesn't the result object already get a _task?

Comment (alikins) [reply lib/ansible/executor/task_executor.py]: It does, but for v2_runner_retry, it's a task before post_validate happens, so the result._task doesnt have the task name templated.

**Comment by alikins** [comment]
@jimi-c updated to include a serializable copy of the Task into '_ansible_result_task' item in the results dict.

**Comment by bcoca** [comment]
do we really want the whole object just to get the name?

**Comment by alikins** [comment]
@bcoca changed to include all of it based on a comment from jimi-c. But looking close it could potentially just include fields that could/did change in post_validate

**Comment by alikins** [comment]
Not sure how to detect when fields would be useful. For a Task(), nothing is always_post_validate. But Base._post_validate excludes Task

**Comment by alikins** [comment]
Updated to take advantages of the changes in 3d65482927e33daa30bb6bfc2153b4c64775ee15

File: test/integration/Makefile [hunk 2]
```diff
@@ -326,3 +326,6 @@ test_lookup_properties:
 
 args:
 	(cd targets/args && ./runme.sh $(TEST_FLAGS))
+
+test_callback_retry_task_name:
```
Comment (sm4rk0) [inline test/integration/Makefile:333]: This won't be run automatically as it's not included in `all`or `other` target. Is this intentional?

Comment (alikins) [reply test/integration/Makefile:333]: Added a test/integration/targets/*/aliases file, so ansible-test will find the tests and run it from CI and locally.

**Pull request merged**